src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -14618,8 +14641,33 @@ uint32 Unit::GetModelForForm(ShapeshiftForm form) const
                         break;
                 }
                 break;
+            }
+            case FORM_AQUATIC_FORM:
+                if (HasAura(114333)) // Glyph of the Orca
+                    return 4591;
```

Comment:
I'm sure no code from TRAVEL_FORM case should be executed if unit does not have this aura